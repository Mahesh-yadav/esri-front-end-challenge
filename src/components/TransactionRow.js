import React from 'react';
import styles from '../styles/TransactionRow.module.css'

function TransactionRow({ transaction, sNo }) {
  return (
    <tr className={styles['row']}>
      <td className={styles['col']}>{sNo}</td>
      <td className={styles['col']}>{transaction['Date']}</td>
      <td className={styles['col']}>{transaction['Transaction Details']}</td>
      <td className={styles['col']}>{transaction['Withdrawal AMT'] ? transaction['Withdrawal AMT'] : '-'}</td>
      <td className={styles['col']}>{transaction['Deposit AMT'] ? transaction['Deposit AMT'] : '-'}</td>
      <td className={styles['col']} >{transaction['Balance AMT']}</td>
    </tr>
  );
}

export default TransactionRow;
