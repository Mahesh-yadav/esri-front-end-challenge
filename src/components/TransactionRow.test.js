import React from 'react';
import ReactDOM from 'react-dom';
import TransactionRow from './TransactionRow';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

describe('Product ', () => {
  const transaction = {
    'Account No': 409000611074,
    Date: '29 Jun 17',
    'Transaction Details': 'TRF FROM  Indiaforensic SERVICES',
    'Value Date': '29 Jun 17',
    'Withdrawal AMT': '',
    'Deposit AMT': '10,00,000.00',
    'Balance AMT': '10,00,000.00',
  };

  test('renders without crashing', () => {
    const tbody = document.createElement('tbody');
    ReactDOM.render(
      <TransactionRow transaction={transaction} sNo={1}></TransactionRow>,
      tbody
    );
  });

  test('renders transaction details inside <td></td>', () => {
    const wrapper = shallow(
      <TransactionRow transaction={transaction} sNo={1}></TransactionRow>
    );

    const details = wrapper.find('.col').at(2);

    expect(details.type()).toBe('td');
    expect(details.text()).toBe(transaction['Transaction Details']);
  });

  test('renders deposit amount correctly when it is not null', () => {
    const wrapper = shallow(
      <TransactionRow transaction={transaction} sNo={1}></TransactionRow>
    );

    const details = wrapper.find('.col').at(4);

    expect(details.type()).toBe('td');
    expect(details.text()).toBe(transaction['Deposit AMT']);
  });

  test('renders withdrawal amount correctly when it is null', () => {
    transaction['Deposit AMT'] = '';

    const wrapper = shallow(
      <TransactionRow transaction={transaction} sNo={1}></TransactionRow>
    );

    const details = wrapper.find('.col').at(4);

    expect(details.type()).toBe('td');
    expect(details.text()).toBe('-');
  });

  test('snapshot', () => {
    const component = renderer.create(
      <TransactionRow transaction={transaction} sNo={1}></TransactionRow>
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
