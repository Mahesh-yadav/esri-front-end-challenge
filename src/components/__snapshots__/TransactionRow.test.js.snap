// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product  snapshot 1`] = `
<tr
  className="row"
>
  <td
    className="col"
  >
    1
  </td>
  <td
    className="col"
  >
    29 Jun 17
  </td>
  <td
    className="col"
  >
    TRF FROM  Indiaforensic SERVICES
  </td>
  <td
    className="col"
  >
    -
  </td>
  <td
    className="col"
  >
    -
  </td>
  <td
    className="col"
  >
    10,00,000.00
  </td>
</tr>
`;
